KomaMRIBase/src/sequences/Grad/design.jl

```diff
@@ -0,0 +1,116 @@
+"""
+"""
+function trapezoid(; duration=0, amplitude=0, area=0, flat_area=0,
+    flat_time=0, rise_time=0, fall_time=0, delay=0,
+    max_grad=Inf, max_slew=Inf, Δtgr=1e-3, sys=nothing)
+
+    if !isnothing(sys)
+        max_grad = sys.Gmax
+        max_slew = sys.Smax
+        Δtgr = sys.GR_Δt
+    end
```

Comment:
This should not be required, set the default for optional parameter sys = Scanner()